### ISSUE TITLE
azuredevops_serviceendpoint_azurecr, unable to update description property

### ISSUE THREAD

**Issue by avishnyakov** [microsoft/terraform-provider-azuredevops#312]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and Azure DevOps Provider) Version

Terraform v0.13.4

```
terraform {
  required_providers {
    azuredevops = {
      source = "microsoft/azuredevops"
      version = "=0.1.2"
    }
  }
}
```

### Affected Resource(s)

* `azuredevops_serviceendpoint_azurecr`

### Terraform Configuration Files

Apply this first
```hcl
resource "azuredevops_serviceendpoint_azurecr" "azurecr" {
  project_id             = data.azuredevops_project.project.id
  service_endpoint_name  = "docker-demo"
  resource_group            = local.docker.demo.rg_name
  azurecr_name              = local.docker.demo.acr_name
  azurecr_spn_tenantid      = local.azure_ad.live.tenant_id
  azurecr_subscription_id   = local.azure.demo.subscription_id
  azurecr_subscription_name = local.azure.demo.subscription_name
}
```

Then try to update description property
```hcl
resource "azuredevops_serviceendpoint_azurecr" "azurecr" {
  project_id             = data.azuredevops_project.project.id
  service_endpoint_name  = "docker-demo"
  resource_group            = local.docker.demo.rg_name
  azurecr_name              = local.docker.demo.acr_name
  azurecr_spn_tenantid      = local.azure_ad.live.tenant_id
  azurecr_subscription_id   = local.azure.demo.subscription_id
  azurecr_subscription_name = local.azure.demo.subscription_name

    # added description
  description = "test description"
}
```

TF plan looks legit

```
      ~ description               = "Managed by Terraform" -> "test description"
```

TF apply fails

```
Error: Error updating service endpoint in Azure DevOps: Updating the endpoint data is not allowed for Azure Container Registry type Docker endpoint. Updated field: 'spnObjectId', Old Value: 'xxxxx-yyyyy-zzzzzx-zzzzzz-uuuuuuuuuuu', New Value: ''.
```

### Expected Behavior

description property can be updated (same as in the UI)

### Actual Behavior

description property CANNOT be updated 

### Steps to Reproduce

1. `terraform apply`

### References

* #0000

**Comment by whiskeysierra** [comment]
I can't offer a fix, but we stopped using `azuredevops_serviceendpoint_azurecr` altogether because it creates a service principal but doesn't manage it afterwards, i.e. it won't get deleted when the service connection gets deleted etc.

We switched to a docker service connection of type "Others" and that works much better:

```hcl
module "service-principal" {
  source = "../service-principal"
  name   = "devops-${var.realm.name}-acr"
}

resource "azurerm_role_assignment" "acr-push" {
  principal_id         = module.service-principal.id
  role_definition_name = "AcrPush"
  scope                = azurerm_container_registry.default.id
}

# Needed because azuredevops_serviceendpoint_azurecr uses the DevOps build service identity to create a service principal in AD.
# But the build service is lacking permissions and since it's a "managed" identity, we can't give it more permissions.
resource "azuredevops_serviceendpoint_dockerregistry" "default" {
  project_id            = var.realm.project.id
  service_endpoint_name = "acr-${var.realm.name}"
  registry_type         = "Others"
  docker_registry       = "https://${azurerm_container_registry.default.login_server}"
  docker_username       = module.service-principal.application.id
  docker_password       = module.service-principal.password
}
```

**Comment by aashishkoirala-ebsco** [comment]
Just hit this today. Looks like the provider is not fetching the spnObjectId from the service connection definition in AzDO before constructing the request that it makes back to AzDO. I just gave up and added `description` to my `ignore_changes` list. I also got another `authorization` property (one that the resource doesn't even support) showing up as changed when working with an `import`ed service connection. I had to add that also to the `ignore_changes` list to make it all work.

**Comment by aashishkoirala** [comment]
Just hit this today. Looks like the provider is not fetching the `spnObjectId` from the service connection definition in AzDO before constructing the request that it makes back to AzDO. I just gave up and added `description` to my `ignore_changes` list. I also got another `authorization` property (one that the resource doesn't even support) showing up as changed when working with an imported service connection. I had to add that also to the `ignore_changes` list to make it all work.
